astropy/units/quantity_helper/function_helpers.py

```diff
@@ -313,7 +313,7 @@ def nan_to_num(x, copy=True, nan=0.0, posinf=None, neginf=None):
         neginf = x._to_own_unit(neginf)
     return (
         (x.view(np.ndarray),),
-        dict(copy=True, nan=nan, posinf=posinf, neginf=neginf),
```

Comment:
I do not like this: Here, `kwargs` is set and I find it much clearer if it looks like arguments to a function.

Reply:
This is a helper function for a numpy operation so speed is very important.
Given that, this is a good improvement.

Reply:
The speed advantage is completely negligible compared to the overhead of even this function itself. I made a conscious decision to write it as it is here, and I don't think this change is good. 

But if you want to speed it up further and keep the intent clear, I'm fine with changing the code to
```
return (x.view(np.ndarray), copy, nan, posinf, neginf), {}, x.unit, None
```

Reply:
I wonder what the speed of constructing a NamedTuple is.